### PR TITLE
GA TTLAfterFinished

### PR DESCRIPTION
**Pull request by sahilvv**
This PR updates the TTLAfterFinish controller to stable as part of 1.23 k8s release.

Related PRs: 
  - https://github.com/kubernetes/kubernetes/pull/105219
  - https://github.com/kubernetes/enhancements/pull/2841

### CODE REVIEW

**Review by sftim — COMMENTED** [review]
This _is_ OK to merge.

Ideally, we should improve this document as well for general availability. I'd reword “The TTL controller only supports Jobs for now” given that we now have a stable definition of what the control plane does.

The comment about NTP could change to be generic over which time sync technology you use.

Nice to have: follow the [style guide](https://kubernetes.io/docs/contribute/style/style-guide/) throughout.

/lgtm

**Comment by sahilvv** [comment]
@sftim @ahg-g reworded to make it explicit that the feature is supported only for jobs. Please let me know if any further changes needed.

File: content/en/docs/concepts/workloads/controllers/ttlafterfinished.md [hunk 1]
```diff
@@ -8,23 +8,17 @@ weight: 70
 
 <!-- overview -->
 
-{{< feature-state for_k8s_version="v1.21" state="beta" >}}
+{{< feature-state for_k8s_version="v1.23" state="stable" >}}
 
 The TTL controller provides a TTL (time to live) mechanism to limit the lifetime of resource
```
Comment (sftim) [inline content/en/docs/concepts/workloads/controllers/ttlafterfinished.md:13]: For GA, I would _prefer_ to rename this to “The TTL-after-finished controller”

for example:
```suggestion
The TTL-after-finished {{<glossary_tooltip text="controller" term_id="controller">}} provides
a TTL (time to live) mechanism to limit the lifetime of Kubernetes workload
```

**Comment by sftim** [comment]
I also recommend looking at https://github.com/kubernetes/website/issues/26917, which is about making sure the right pages hyperlink to the TTLAfterFinished docs.

**Comment by sahilvv** [comment]
@sftim I have addressed the review feedback given so far. Please take a look. Thanks again!

File: content/en/docs/concepts/workloads/controllers/ttlafterfinished.md [hunk 2]
```diff
@@ -62,12 +57,12 @@ returns a successful API response.
...
 
-Because TTL controller uses timestamps stored in the Kubernetes resources to
+Because TTL-after-finished controller uses timestamps stored in the Kubernetes resources to
 determine whether the TTL has expired or not, this feature is sensitive to time
 skew in the cluster, which may cause TTL controller to clean up resource objects
 at the wrong time.
 
-In Kubernetes, it's required to run NTP on all nodes
+In Kubernetes, it's required to run a time sync solution on all nodes
 (see [#6159](https://github.com/kubernetes/kubernetes/issues/6159#issuecomment-93844058))
```
Comment (sftim) [inline content/en/docs/concepts/workloads/controllers/ttlafterfinished.md:66]: nit: I'd omit this line.

Comment (sahilvv) [reply content/en/docs/concepts/workloads/controllers/ttlafterfinished.md]: removed @sftim.

File: content/en/docs/concepts/workloads/controllers/ttlafterfinished.md [hunk 3]
```diff
@@ -1,36 +1,31 @@
...
+have finished execution. TTL controller only handles 
+{{< glossary_tooltip text="Jobs" term_id="job" >}}.
 
 <!-- body -->
 
-## TTL Controller
+## TTL-after-finished Controller
 
-The TTL controller only supports Jobs for now. A cluster operator can use this feature to clean
+The TTL-after-finished controller is only supported for Jobs. A cluster operator can use this feature to clean
```
Comment (janetkuo) [inline content/en/docs/concepts/workloads/controllers/ttlafterfinished.md:22]: It we change this, we should also change other references of "resources" to "Jobs".

Comment (sahilvv) [reply content/en/docs/concepts/workloads/controllers/ttlafterfinished.md:22]: Great comment @janetkuo. I have updated the references of `resources` to `jobs`.
